modules/drivers/bigquery/src/metabase/driver/bigquery/query_processor.clj

```diff
@@ -31,14 +31,15 @@
 
 ;; TODO -- I think this only applied to Fields now -- see
 ;; https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language. It definitely doesn't apply
-;; to Tables. Not sure about project/dataset identifiers.
+;; to Tables. Datasets can be passed as `dataset-id` or `project-id`.`dataset-id`.
+;; TODO -- Needs to change frontend too
 (defn- valid-bigquery-identifier?
   "Is String `s` a valid BigQuery identifier? Identifiers are only allowed to contain letters, numbers, and underscores;
   cannot start with a number; and can be at most 128 characters long."
   [s]
   (boolean
    (and (string? s)
-        (re-matches #"^([a-zA-Z_][a-zA-Z_0-9]*){1,128}$" s))))
+        (re-matches #"^([a-zA-Z_\-.][a-zA-Z_0-9]*){1,128}$" s))))
```

Comment:
Hi @viniciusdsmello 

I'm guessing this is trying to fix #13245

The punctuation is not escaped, so it's essentially a wildcard (any) character.

Seems like datasets (https://cloud.google.com/bigquery/docs/datasets) allows `a-zA-Z_0-9`, but projects are a slightly more complex rule https://cloud.google.com/resource-manager/docs/creating-managing-projects

And not sure if a combined identifier then would have a different max length. Do you have any documentation about that?

You might want to have a look at #13111 too.

Reply:
Please see #13111 for reasons for the length. And there should probably be tests too.

But the regex is also only applying to the first character. The second bracket is for all the other characters.

Reply:
I've included the changes proposed at #13111 into this PR and added the test for this new regex. Since project name can have up to 30 characters and datasets up to 1024, I put the sum as maximum length.